### PR TITLE
Add New Features page to patterns

### PR DESCRIPTION
**Pull request by heyjoethomas**
Moving the new features/new badge pattern back into the patterns section and out of legacy. New content has also been added for clarity. 

Screenshot 2023-05-10 at 2 15 22 PM

### WHY are these changes introduced?
There has been an increasing need for guidance on how we communicate new features in the admin to merchants. Publishing this content is a stop-gap measure as we collect more comprehensive thoughts on the topic across teams. 

Fixes #0000 

### How to 🎩

🖥 [Local development instructions](https://github.com/Shopify/polaris/blob/main/README.md#local-development)
🗒 [General tophatting guidelines](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md)
📄 [Changelog guidelines](https://github.com/Shopify/polaris/blob/main/.github/CONTRIBUTING.md#changelog)

<details>
<summary>Copy-paste this code in <code>playground/Playground.tsx</code>:</summary>

```jsx
import React from 'react';
import {Page} from '../src';

export function Playground() {
  return (
    <Page title="Playground">
      {/* Add the code you want to test in here */}
    </Page>
  );
}
```

</details>

### 🎩 checklist

- [ ] Tested on [mobile](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md#cross-browser-testing)
- [ ] Tested on [multiple browsers](https://help.shopify.com/en/manual/shopify-admin/supported-browsers)
- [ ] Tested for [accessibility](https://github.com/Shopify/polaris/blob/main/documentation/Accessibility%20testing.md)
- [ ] Updated the component's `README.md` with documentation changes
- [ ] [Tophatted documentation](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting%20documentation.md) changes in the style guide

### CODE REVIEW

File: polaris.shopify.com/content/patterns/new-features/index.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
+---
+title: New Features
```
Comment (sam-b-rose) [inline polaris.shopify.com/content/patterns/new-features/index.md]:
```suggestion
title: New features
```

**Review by sam-b-rose — APPROVED** [review]
Thanks for bringing back this legacy new feature!?! 💯  Just one small casing-related comment.

**Comment by sophschneider** [comment]
@johanstromqvist Is it ok to move this out of legacy when it doesn't have the same formatting as the current pattern pages?

**Comment by johanstromqvist** [comment]
Love the quick response!

A few things to know
- "Legacy patterns" is no longer a concept on the doc site. The `/patterns-legacy/` part of the URL is residue that is yet to be dealt with.
- We currently place the old pages that don't follow the new convention in "Archived documentation" and make it clear that they are still useful. This is a bit hacky, but a quick fix for an ephemeral problem.
- The reason the old pages are not showing up in the nav is because of a [technical constraint](https://github.com/Shopify/polaris/issues/8949#issuecomment-1520546669) that we don't think is worth solving given that these four pages will be ported to the primary index.

image

If you're making changes to the pattern index or the pattern IA, please ping me or one of the engineer from the patterns team as we're working actively on these areas, so that we can avoid double work or clashing directions.

> @johanstromqvist Is it ok to move this out of legacy when it doesn't have the same formatting as the current pattern pages?

@sophschneider that would break the convention we're working with and make it less clear what content and features to expect from the pattern pages. I'd recommend to keep them in the second group for now and instead iterate on the headline and copy that separates them.

This is a temporary problem on the doc site, so I'm hesitant to invest too much in it.

File: polaris.shopify.com/content/patterns/new-features/index.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+<div as="Variants"></div>
+
+<div as="Stack" gap="4">
+
+## Related resources
+
+- See another two-column layout in use in the [Resource detail layout](/patterns/resource-details-layout) pattern.
+- See a single-column layout in use in the [Resource index layout](/patterns/resource-index-layout) pattern.
+- Learn more about [Layout](https://shopify.dev/apps/design-guidelines/layout) in the app design guidelines.
+- Check out the Polaris [Space](/design/space) guidelines to understand Polaris grid and spacing scale.
```
Comment (sophschneider) [inline polaris.shopify.com/content/patterns/new-features/index.md]: Are layouts or space is related to new feature? or is this copied over from somewhere else?

**Comment by sophschneider** [comment]
@johanstromqvist @heyjoethomas what do you think about having this page live under one of `Foundations`, `Design`, or `Content` until we re-work patterns? The pattern section is going to be working through some major IA changes


Screenshot 2023-05-10 at 4 23 28 PM

File: polaris.shopify.com/content/patterns/new-features/variants/default.md [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+## When to highlight new features
+
+Features should have all 3 of the following characteristics before highlighting them to merchants with a New Badge or PIP:
+
+- Shopify wants to drive adoption of this feature because it has high business value
+- It’s creating new and outsized value for merchants (not just an improved way of doing something they are already doing)
+- It’s worth distracting merchants from their current workflow to inform them about a whole new part of Shopify
+
+Things that generally meet this criteria are large and impactful. Examples could be entire new pages in settings or the introduction of an entirely new technology or capability.
```
Comment (martenbjork) [inline polaris.shopify.com/content/patterns/new-features/variants/default.md:13]: I think this is well written, but I fear that teams might have a bit of "tunnel vision" about their feature and think that it ticks all these boxes. Should we make it even stronger? 

`Features should rarely have "new" indicators` or something like that.

File: polaris.shopify.com/content/patterns/new-features/variants/default.md [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+- Shopify wants to drive adoption of this feature because it has high business value
+- It’s creating new and outsized value for merchants (not just an improved way of doing something they are already doing)
+- It’s worth distracting merchants from their current workflow to inform them about a whole new part of Shopify
+
+Things that generally meet this criteria are large and impactful. Examples could be entire new pages in settings or the introduction of an entirely new technology or capability. 
+
+## Avoid highlighting individual features within a page
+
+As the Shopify admin UI continues to evolve, it's natural for teams to want to draw attention to new features and improvements. However, we strongly advise against using "new" badges or blue pips to highlight these additions within individual sections like Products, Orders, and Customers.
+
```
Comment (martenbjork) [inline polaris.shopify.com/content/patterns/new-features/variants/default.md:20]: I think this should live before the previous section. This makes a good intro.

File: polaris.shopify.com/content/patterns/new-features/variants/default.md [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+
+### Why this should be avoided
+
+  **Inconsistency** - Different teams might use different methods to highlight features, leading to an inconsistent experience for merchants.
+  
+  **Visual clutter** - Over time, the admin UI may become cluttered with numerous badges and pips, making it difficult for merchants to focus on their primary tasks.
+
```
Comment (martenbjork) [inline polaris.shopify.com/content/patterns/new-features/variants/default.md:26]: Formatting: This should probably just be h4s (####)

File: polaris.shopify.com/content/patterns/new-features/variants/default.md [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+## When to highlight new features
+
+Features should have all 3 of the following characteristics before highlighting them to merchants with a New Badge or PIP:
+
+- Shopify wants to drive adoption of this feature because it has high business value
+- It’s creating new and outsized value for merchants (not just an improved way of doing something they are already doing)
+- It’s worth distracting merchants from their current workflow to inform them about a whole new part of Shopify
+
+Things that generally meet this criteria are large and impactful. Examples could be entire new pages in settings or the introduction of an entirely new technology or capability. 
+
```
Comment (martenbjork) [inline polaris.shopify.com/content/patterns/new-features/variants/default.md:16]: There is a risk that teams have tunnel vision with regards to their new feature, and think that their feature ticks all these boxes. I think we can phrase this even more strongly.

**Review by martenbjork — APPROVED** [review]
This is great! Really glad you're jumping on this 🔥 I left some comments, but no blockers.

File: polaris.shopify.com/content/patterns/new-features/index.md (hunk 2 above)
Comment (heyjoethomas) [reply polaris.shopify.com/content/patterns/new-features/index.md]: Yes. My bad. Will remove

File: polaris.shopify.com/content/patterns/new-features/variants/default.md (hunk 5 above)
Comment (heyjoethomas) [reply polaris.shopify.com/content/patterns/new-features/variants/default.md:26]: The spacing on the type isn't great this way which is why I had it formatted the way I did. Thoughts? 
Screenshot 2023-05-11 at 9 57 32 AM

**Comment by heyjoethomas** [comment]
> @johanstromqvist @heyjoethomas what do you think about having this page live under one of `Foundations`, `Design`, or `Content` until we re-work patterns? The pattern section is going to be working through some major IA changes
> 
> Screenshot 2023-05-10 at 4 23 28 PM

I don't think that makes sense. Conceptually this is a pattern but I understand the formatting issues. This should be temporary however. There are already meetings on the books to help figure out a more comprehensive solution. At that point, we would certainly revisit this and bring it into alignment with the rest of the section.

File: polaris.shopify.com/content/patterns/new-features/variants/default.md (hunk 6 above)
Comment (heyjoethomas) [reply polaris.shopify.com/content/patterns/new-features/variants/default.md:16]: Good idea. Will shift this around.

File: polaris.shopify.com/content/patterns/new-features/variants/default.md [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+### Why this should be avoided
+
+ #### Inconsistency
+ 
+ Different teams might use different methods to highlight features, leading to an inconsistent experience for merchants.
```
Comment (martenbjork) [inline polaris.shopify.com/content/patterns/new-features/variants/default.md]: There is a stray space character before this paragraph and the #### before it.

Comment (heyjoethomas) [reply polaris.shopify.com/content/patterns/new-features/variants/default.md]: 🤦‍♂️

File: polaris.shopify.com/content/patterns/new-features/variants/default.md (hunk 5 above)
Comment (martenbjork) [reply polaris.shopify.com/content/patterns/new-features/variants/default.md:26]: Crikey, that's awful but I'd still ship the h4s. Avoiding the h4s only hides the issue and prevents it from being fixed.

**Comment by johanstromqvist** [comment]
I strongly suggest to keep the page in the secondary list. We are working hard to establish new expectations among builders what solutions to find in pattern documentation and why to go there, Moving this page to the primary list would muddy that. 

The only tradeoff that I can see in keeping the page in the secondary list is that it won't show up in the left nav, and that's a temporary problem that will disappear when we've ported the old pages to the new format.

Like I suggested above, we can improve the section headline and copy to make the pages in that section feel more legitimate. "Archived documentation" does have some negative connotation to it.

Recommendation
- [ ] Place the page in the second list
- [ ] Update headline to `Additional guidelines` (or similar, open to suggestions)
- [ ] Update the copy to `Find best practices and design guidelines to common merchant needs` (or similar, open to suggestions)

**Comment by heyjoethomas** [comment]
Hey @johanstromqvist couple questions since I don't have as much context here: 

> We are working hard to establish new expectations among builders what solutions to find in pattern documentation and why to go there, Moving this page to the primary list would muddy that.

Can you expound on this? 

> that's a temporary problem that will disappear when we've ported the old pages to the new format

Do we have an ETA on this? 

> best practices and design guidelines to common merchant needs

Isn't that the definition of a pattern? If not, how would you explain the different between a pattern and guidance to common merchant needs?

**Comment by johanstromqvist** [comment]
> Hey @johanstromqvist couple questions since I don't have as much context here:
> 
> > We are working hard to establish new expectations among builders what solutions to find in pattern documentation and why to go there, Moving this page to the primary list would muddy that.
> 
> Can you expound on this?
> 
> > that's a temporary problem that will disappear when we've ported the old pages to the new format
> 
> Do we have an ETA on this?
> 
> > best practices and design guidelines to common merchant needs
> 
> Isn't that the definition of a pattern? If not, how would you explain the different between a pattern and guidance to common merchant needs?

Great Qs! Let's limit this PR to the content update and chat about the general direction and intent of the section separately. Would love your feedback and critique on it.

Also created this issue #9205

**Comment by themarkappleby** [comment]
Even though this page is publicly accessible, the copy here seems to be written with Shopifolk as the intended audience. Was this intentional? It was a little unexpected. For example:

> we strongly advise against using "new" badges or blue pips to highlight these additions within individual sections like Products, Orders, and Customers

AFAIK there is currently no mechanism by which a 3P partner could add new UI to products, orders, or customers pages.

> Shopify wants to drive adoption of this feature because it has high business value

This doesn't seem applicable if I am a 3P app developer reading this page.

> Examples could be entire new pages in settings

Again, AFAIK it isn't possible for a 3P partner to create an entirely new page in settings.

**Comment by jeffkraemer2** [comment]
@heyjoethomas I think @themarkappleby raises good points above. If this page is publicly accessible, we should revisit the content that's Shopify-specific. Also, I have a team member @kristyyy-chen that's looking at a new feature awareness system for Admin, i.e. for an internal audience only, so let's compare notes on what should be in Polaris and what should be in the Vault.

**Comment by johanstromqvist** [comment]
The updated page also ended up in the wrong section, as opposed to what we agreed on. And the old page is still listed, but throws a 404.

**Pull request merged**